src/Lecture3.hs

```diff
@@ -92,8 +98,13 @@ weekday to the second.
 >>> daysTo Friday Wednesday
 5
 -}
-daysTo = error "TODO"
-
+daysTo :: (Eq a, Bounded a, Enum a) => a -> a -> Int
+daysTo = go 0
+  where
+    go !count !first second
+      | first == second = count
+      | otherwise = go (count + 1) (next first) second
```

Comment:
That's a correct implementation! 👍🏻 
You can make it more efficient and avoid recursion by using math operations 🔢
Although, such implementation would make more sense for a Weekday-specific function 🙂